pyHS100/pyHS100.py

```diff
@@ -285,6 +268,156 @@ def identify(self):
 
         return alias, model, features
 
+    @property
+    def alias(self):
+        """
+        Get current device alias (name)
+        :return:
+        """
+        return self._alias
+
+    @alias.setter
+    def alias(self, alias):
+        """
+        Sets the device name aka alias.
+        :param alias: New alias (name)
+        """
+
+        self._query_helper("system", "set_dev_alias", {"alias": alias})
+
+    @property
+    def led(self):
+        """
+        Returns the state of the led.
+        :return:
+        """
+        return {"led": 1 - self.sys_info["led_off"]}
+
+    @led.setter
+    def led(self, state):
+        """
+        Sets the state of the led (night mode)
+        :param state: 1 to set led on, 0 to set led off
+        """
+
+        self._query_helper("system", "set_led_off", {"off": 1 - state})
+
+    @property
+    def icon(self):
+        """
+        Returns device icon
+        Note: this doesn't seem to work when not using the cloud service, not tested with it either.
+        :return:
+        """
+        return self._query_helper("system", "get_dev_icon")
+
+    @icon.setter
+    def icon(self, icon):
+        """
+        Content for hash and icon are unknown.
+        :param icon:
+        """
+
+        raise SmartPlugNotImplementedException()
```

Comment:
I'm not sure there is much of a benefit to roll our own Exception in this case. I'd say lets simply raise `NotImplementedError`.